### PR TITLE
Duplicate method inspection

### PR DESCRIPTION
**Pull request by linniksa**
Simple implementation
relates issue(s): #1023

### CODE REVIEW

File: src/com/goide/inspections/GoDuplicateMethodInspection.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import java.util.Collection;
+import java.util.List;
+
+
+public class GoDuplicateMethodInspection extends GoInspectionBase {
+  @Override
+  protected void checkFile(@NotNull GoFile file, @NotNull final ProblemsHolder problemsHolder) {
+    final MultiMap<String, GoMethodDeclaration> map = new MultiMap<String, GoMethodDeclaration>();
+    List<GoFile> files = GoPsiImplUtil.getAllPackageFiles(file);
+    for (GoFile goFile : files) {
```
Comment (zolotov) [inline src/com/goide/inspections/GoDuplicateMethodInspection.java]: Please use GoMethodIndex instead of constructing your own map.

**Comment by linniksa** [comment]
> Please use GoMethodIndex instead of constructing your own map.

done, also `GoDuplicateFunctionInspection` should be updated to use `GoFunctionIndex`?

**Comment by zolotov** [comment]
Merged, thanks.

> done, also GoDuplicateFunctionInspection should be updated to use GoFunctionIndex?

Yes, it would be nice. Also I hope @ignatov will share his thought about global duplications inspection

**Comment by ignatov** [comment]
Yep, we should unify these inspection in a single one.

**Comment by linniksa** [comment]
@ignatov 
something like that: https://github.com/linniksa/go-lang-idea-plugin/commit/ba7be67945fcf2eab69b9599c8d04f5d87a5a552 ?

**Pull request merged**